src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    public static final int INITIAL_CAPACITY = 10;
+    public static final String NO_SUCH_ELEMENT_MESSAGE = "There is no such element in the list - ";
+    public static final String OUT_OF_BOUNDS_MESSAGE = "Index out of bounds";
+
+    private Object[] storage;
+    private int size = 0;
+
+    public ArrayList() {
+        storage = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index - 1 == size) {
+            add(value);
+        } else {
+            ensureCapacity();
+            ensureAddition(index);
+            Object[] temp = new Object[storage.length];
+            System.arraycopy(storage, 0, temp, 0, index);
+            temp[index] = value;
+            System.arraycopy(storage, index, temp, index + 1, size - index);
+            storage = temp;
```

Comment:
we don't need `temp` variable, we can swap 34 and 35 lines and after that use `storage` instead of `temp`